### PR TITLE
net.davidotek.pupgui2

### PR DESCRIPTION
**Pull request by DavidoTek**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.
- [X] I am an upstream contributor to the project.
- [X] I own the domain used in the application ID.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

#### Permissions
The app needs access to the Wine/Proton compatibility tools folders of [Steam](https://store.steampowered.com/), [Lutris](https://github.com/lutris/lutris/) and [Heroic Games Launcher](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher).

Folder|Reason
---|---
\~/.steam/root/compatibilitytools.d<br>\~/.local/share/Steam/compatibilitytools.d|Steam compatibility tools folder
\~/.var/app/com.valvesoftware.Steam/data/Steam/compatibilitytools.d|Steam Flatpak compatibility tools folder
\~/.steam/root/config:ro<br>\~/.local/share/Steam/config:ro<br>\~/.var/app/com.valvesoftware.Steam/data/Steam/config:ro|Steam config containing games<br>using the compatibility tools
\~/.local/share/lutris/runners/wine|Lutris compatibility tools folder
\~/.config/heroic/tools|Heroic compatibility tools folder

#### Additional Patches
Upstream uses Qt6/PySide6 but there is only a KDE runtime for Qt5. Bundling Qt6 is unnecessary.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by DavidoTek** [comment]
bot, build net.davidotek.pupgui2

File: net.davidotek.pupgui2.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+app-id: net.davidotek.pupgui2
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline net.davidotek.pupgui2.yml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

Comment (hfiguiere) [reply net.davidotek.pupgui2.yml:3]: or use `6.2` for Qt6.

File: net.davidotek.pupgui2.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
```
Comment (hfiguiere) [inline net.davidotek.pupgui2.appdata.xml:2]: was this submitted upstream?

File: net.davidotek.pupgui2.desktop [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline net.davidotek.pupgui2.desktop:1]: was this submitted upstream?

Comment (DavidoTek) [reply net.davidotek.pupgui2.desktop:1]: Upstream doesn't use a desktop file as it is generated automatically by AppImageBuilder. Could include it in upstream though.

File: net.davidotek.pupgui2.appdata.xml (hunk 2 above)
Comment (DavidoTek) [reply net.davidotek.pupgui2.appdata.xml:2]: There seems to be a bug with AppImageBuilder that the AppStream can't contain a `launchable` tag for some reason.

File: net.davidotek.pupgui2.yml (hunk 1 above)
Comment (DavidoTek) [reply net.davidotek.pupgui2.yml:3]: will try it

File: net.davidotek.pupgui2.desktop (hunk 3 above)
Comment (hfiguiere) [reply net.davidotek.pupgui2.desktop:1]: it should be generated/installed by the standard build process. it is specific to neither. a "destkop" app without a `.desktop` file is a bug.

Comment (DavidoTek) [reply net.davidotek.pupgui2.desktop:1]: Will include it in upstream

File: net.davidotek.pupgui2.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://davidotek.github.io/protonup-qt/images/screenshot143_dark1.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://davidotek.github.io/protonup-qt/images/screenshot143_light1.png</image>
+    </screenshot>
+  </screenshots>
+
```
Comment (hfiguiere) [inline net.davidotek.pupgui2.appdata.xml:27]: missing `releases` and `content_rating`

**Comment by DavidoTek** [comment]
bot, build net.davidotek.pupgui2

**Comment by hfiguiere** [comment]
bot, build net.davidotek.pupgui2

**Comment by DavidoTek** [comment]
> Build [69374 failed](https://flathub.org/builds/#/builders/32/builds/69374)

For some reason it fails to build with clang with `5.15-21.08`. Same goes for `6.2`... `5.15` is fine.

**Comment by DavidoTek** [comment]
bot, build net.davidotek.pupgui2

**Comment by DavidoTek** [comment]
bot, build net.davidotek.pupgui2

File: net.davidotek.pupgui2.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.llvm12"
+    ],
+    "command": "net.davidotek.pupgui2",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--persist=.config/pupgui",
```
Comment (barthalion) [inline net.davidotek.pupgui2.json:15]: This shouldn't be needed, as long as your application respects $XDG_CONFIG_HOME variable.

**Comment by DavidoTek** [comment]
bot, build net.davidotek.pupgui2

**Pull request closed without merging**